### PR TITLE
Add logging to to redis statsd python script, also add try/catch error 

### PR DESCRIPTION
**Pull request by manansaraf**
### SUMMARY
* [GA-45211](https://jira.dp.hbo.com/browse/GA-45211)
 ### REVIEWERS
- [ ] @HBOCodeLabs/content-services
  ### DETAILS
Added try/catch around the python script so that the script does not crash
 ### CUSTOMER IMPACT
This file is only used to send messages from redis to our statsd aggregator
 ### DEVELOPER IMPACT
None, except better logging for the script

### PIC
image

### CODE REVIEW

File: redis-statsd.py [hunk 1]
```diff
@@ -102,56 +105,70 @@ def linesplit(socket):
...
+                    send_metric('{}.keyspace.{}'.format(
+                        args.prefix, kg), 'g',
+                        float(stats['keyspaces'][ks][kg]), ['keyspace={}'.format(ks)]
+                    )
+                    logger.info('Metric sent: {}.keyspace.{}'.format(
+                        args.prefix, kg), 'g',
+                        float(stats['keyspaces'][ks][kg]), ['keyspace={}'.format(ks)]
+                    )
+
+        out_sock.close()
```
Comment (matteraByte) [inline redis-statsd.py:158]: Should probably define out_sock outside of `try` block and close in `finally` block to ensure we also close a possible open connection.

**Review by jlutzhbo — COMMENTED** [review]
Spoke offline about this. The logging is great, but I would recommend that an error crash the script, since it may not be recoverable, and we should rely on the OS to respawn it when it crashes.

**Review by jlutzhbo — CHANGES_REQUESTED** [review]
Spoke offline about this. The logging is great, but I would recommend that an error crash the script, since it may not be recoverable, and we should rely on the OS to respawn it when it crashes.

**Review by brianlunderhbo — CHANGES_REQUESTED** [review]
Agree with Jon and Chris. Also, this will be logging tons of messages, by my count up to 28 every 10 ms or 4,032,000 lines per day. That is probably too many. Maybe we can consolidate the logs so that it only logs with errors and during normal operation it maybe logs something like "reported # of metrics in the past # seconds" every once in a while.

**Comment by dlang-src** [comment]
Sorry for jumping in here (owner of logging and metrics).  Is this going to run on the EC2 instances for redis in us-east-1?   If so, could I convince you to use a prometheus [exporter](https://github.com/oliver006/redis_exporter) instead of trying to use a statsd output?

File: redis-statsd.py [hunk 2]
```diff
@@ -15,6 +16,8 @@
 
 args = parser.parse_args()
 
+logging.basicConfig(filename='/var/log/redis-statsd.log', filemode='a', format='%(levelname)s - %(message)s')
```
Comment (luisgarciaalanis) [inline redis-statsd.py:19]: filemode='a'? why not 'r'?

File: redis-statsd.py [hunk 3]
```diff
@@ -102,56 +105,60 @@ def linesplit(socket):
...
+                if kg in stats['keyspaces'][ks]:
+                    send_metric('{}.keyspace.{}'.format(
+                        args.prefix, kg), 'g',
+                        float(stats['keyspaces'][ks][kg]), ['keyspace={}'.format(ks)]
+                    )
+        out_sock.close()
+        time.sleep(10)
+    except Exception, e:
+        logger.error('Error occured! Message: {} Args: {}'.format(e.message, e.args))
+        exit(0)
```
Comment (luisgarciaalanis) [inline redis-statsd.py]: exit(1)

**Review by luisgarciaalanis — APPROVED** [review]

**Review by brianlunderhbo — APPROVED** [review]

File: redis-statsd.py (hunk 2 above)
Comment (jlutzhbo) [inline redis-statsd.py:19]: nit: Should we pass the log file location as an argument?

**Review by jlutzhbo — APPROVED** [review]
Looks good, just one question.

File: redis-statsd.py (hunk 2 above)
Comment (manansaraf) [reply redis-statsd.py:19]: We would have to change how this python file is initialized which will another PR, I just want to fix this, afterwards I can change all the initialization to move it out

**Pull request merged**